cpp/src/utils/metrics/yaml_config.hh

```diff
@@ -0,0 +1,38 @@
+#ifndef TS_CPP_UTILS_METRICS_YAML_CONFIG_HH_
+#define TS_CPP_UTILS_METRICS_YAML_CONFIG_HH_
+
+#include <yaml-cpp/yaml.h>
+
+#include <set>
+#include <string>
+#include <vector>
+
+#include "src/utils/metrics/config.hh"
+#include "src/utils/metrics/metric.hh"
+
+namespace torchserve {
+class YAMLMetricsConfigurationHandler : public MetricsConfigurationHandler {
+ public:
+  YAMLMetricsConfigurationHandler()
+      : dimension_names{}, system_metrics{}, model_metrics{} {}
+  ~YAMLMetricsConfigurationHandler() override {}
+  void LoadConfiguration(const std::string metrics_config_file_path) override;
+  std::set<std::string> GetDimensionNames() override;
+  std::vector<MetricConfiguration> GetSystemMetrics() override;
+  std::vector<MetricConfiguration> GetModelMetrics() override;
```

Comment:
the cost is high to return std::set or std::vector.

Reply:
Given that this method is likely to be called only once during initialization of the metrics cache, would it be reasonable to make this tradeoff in the interest of keeping the implementation simple?